test/integration/integration_test.cc

```diff
@@ -126,6 +135,11 @@ TEST_P(IntegrationTest, AdminDrainDrainsListeners) {
 
   // Validate that the listeners have been stopped.
   test_server_->waitForCounterEq("listener_manager.listener_stopped", 1);
+
+  Network::TcpListenSocket socket(
```

Comment:
can you probably close the existing `codec_client_` and create  a fresh connection with  `makeHttpConnection(lookupPort("http"))` and see if it fails immediately instead of trying to create a socket because that is how it is used generally?

Reply:
Just tried it, but it doesn't look like even `makeHeaderOnlyRequest` fails, and `waitForNextUpstreamRequest` would imply timeout

Reply:
Ok.  Thanks, Then probably we should add `EXPECT_NO_THROW` here to validate that it is actually not throwing any exceptions = @mattklein123 any other ideas?

Reply:
I think this is fine (it would fail if it throws so it's OK but a comment might be good). I think you could also just make a raw TCP connection (see the TCP proxy integration tests) and make sure that fails?

Reply:
+1 for rawTCP connection. Was suggesting EXPECT_NO_THROW because it is self explanatory